### ISSUE TITLE
Shapes load empty generic parameters

### ISSUE THREAD

**Issue by palvarezlopez** [eclipse/sumo#4428]
In SUMO and Netedit. It's a problem of ShapeHandler.

**Comment by namdre** [comment]
That was a feature. Shapes with empty parameter values are sometimes imported from OSM. 
Since generic parameters are end user-defined they should allow for more freedoms.

**Comment by palvarezlopez** [comment]
@namdre this include non valid characters?

**Comment by namdre** [comment]
Yes. Everything that can be loaded technically from XML is allowed.

**Comment by palvarezlopez** [comment]
@namdre But the key cannot be empty, because the following case results in an inconsistency:

```
<poly id=.......>
  <parameter key="" value ="A"/>
  <parameter key="" value ="B"/>
</poly>
```

Generic Parameters are saved in a std::map. These map uses the Generic Parameter 'key' as key, (forgive the redundancy), then in this case only "B" will be written because it will overwrite the last inserted parameter "A"

**Comment by namdre** [comment]
You are right. We should not allow empty keys. (And I never saw one. Only empty values).
